src/aioca/_catools.py

```diff
@@ -539,20 +538,24 @@ async def __create_subscription(
             self._as_parameter_ = event_id.value
 
             # If we are async then we should be servicing the callbacks here
-            while self.__is_async is not False:
-                try:
-                    with self.__lock:
-                        value = self.pending_values.popleft()
-                except IndexError:
+            while not self.__is_sync:
+                with self.__lock:
+                    # Consume all the values from the queue
+                    values = [
+                        self.pending_values.popleft()
+                        for _ in range(len(self.pending_values))
+                    ]
```

Comment:
Is there a reason for doing it this way rather than just `values = list(self.pending_values)`?